include/arch/mips/danube/linker.ld

```diff
@@ -0,0 +1,352 @@
+/*
+ * A platform and target independent link script to produce UHI
+ * compliant binaries with varying levels of system initialization
+ * support.
+ */
+
+ #define _LINKER
+ #define _ASMLANGUAGE
+
+ #include <autoconf.h>
+ #include <linker/sections.h>
+
+ #include <linker/linker-defs.h>
+ #include <linker/linker-tool.h>
+
+ #ifdef CONFIG_XIP
+ #define ROMABLE_REGION              ROM
+ #else
+ #define ROMABLE_REGION              RAM
+ #endif
+ #define RAMABLE_REGION              RAM
+
+ MEMORY
+ {
+     ROM (rx)  : ORIGIN = CONFIG_MIPS_ROM_BASE_ADDR, LENGTH = CONFIG_MIPS_ROM_SIZE
+     RAM (rwx) : ORIGIN = CONFIG_MIPS_RAM_BASE_ADDR, LENGTH = CONFIG_MIPS_RAM_SIZE
+     /* Used by and documented in include/linker/intlist.ld */
+     IDT_LIST  (wx)      : ORIGIN = 0xFFFFF7FF, LENGTH = 2K
+ }
+
+__entry = DEFINED(__reset_vector) ? MIPS_ROM_BASE_ADDR : _start;
```

Comment:
what's the story with this? why do we need to conditionally define the entry point?